eva/executor/delete_executor.py

```diff
@@ -34,53 +33,35 @@ def __init__(self, node: ProjectPlan):
-
-            del_batch = Batch()
+        table_catalog = self.node.table_ref.table.table_obj
+        storage_engine = StorageEngine.factory(table_catalog)
+        del_batch = Batch()
 
-            if table_catalog.table_type == TableType.VIDEO_DATA:
-                raise NotImplementedError("DELETE only implemented for structured data")
-            elif table_catalog.table_type == TableType.IMAGE_DATA:
-                raise NotImplementedError("DELETE only implemented for structured data")
-            elif table_catalog.table_type == TableType.STRUCTURED_DATA:
-                del_batch = storage_engine.read(table_catalog)
-                del_batch = list(del_batch)[0]
+        assert (
+            table_catalog.table_type == TableType.STRUCTURED_DATA
+        ), "DELETE only implemented for structured data"
 
-            # Added because of inconsistency in col_alias in Structured data Batch project function
-            original_column_names = list(del_batch.frames.columns)
-            column_names = [
-                f"{table_catalog.name.lower()}.{name}"
-                for name in original_column_names
-                if not name == "_row_id"
-            ]
-            column_names.insert(0, "_row_id")
-            del_batch.frames.columns = column_names
-            del_batch = apply_predicate(del_batch, self.predicate)
+        del_batch = storage_engine.read(table_catalog)
+        del_batch = list(del_batch)[0]
 
-            # All the batches that need to be deleted
+        # Added because of inconsistency in col_alias in Structured data Batch project function
+        original_column_names = list(del_batch.frames.columns)
+        column_names = [
+            f"{table_catalog.name.lower()}.{name}"
+            for name in original_column_names
+            if not name == "_row_id"
+        ]
+        column_names.insert(0, "_row_id")
+        del_batch.frames.columns = column_names
+        del_batch = apply_predicate(del_batch, self.predicate)
 
-            if table_catalog.table_type == TableType.VIDEO_DATA:
-                storage_engine.delete(table_catalog, del_batch)
-            elif table_catalog.table_type == TableType.IMAGE_DATA:
-                storage_engine.delete(table_catalog, del_batch)
-            elif table_catalog.table_type == TableType.STRUCTURED_DATA:
-                del_batch.frames.columns = original_column_names
-                table_needed = del_batch.frames[
-                    [f"{self.predicate.children[0].col_name}"]
-                ]
-                for num in range(len(del_batch)):
-                    storage_engine.delete(table_catalog, table_needed.iloc[num])
-            yield Batch(pd.DataFrame(["Deleted row"]))
+        # All the batches that need to be deleted
 
-        except Exception as e:
-            logger.error(e)
-            raise ExecutorError(e)
+        if table_catalog.table_type == TableType.STRUCTURED_DATA:
```

Comment:
We do not need this if check, the previous `assert` ensures that `table_catalog.table_type` is `TableType.STRUCTURED_DATA`